value/src/main/java/com/google/auto/value/processor/autovalue.vm

```diff
@@ -185,4 +185,105 @@ final class $subclass$formalTypes extends $origClass$actualTypes {
 #if (!$serialVersionUID.empty)
   private static final long serialVersionUID = $serialVersionUID;
 #end
+
+#if ($builderTypeName != "")
+
+  #foreach ($m in $toBuilderMethods)
+
+  @Override
+  public ${builderTypeName}${builderActualTypes} ${m}() {
+    return new Builder${builderActualTypes}(this);
+  }
+
+  #end
+
+  static final class Builder${builderFormalTypes} ##
```

Comment:
+`private`? This type isn't used anywhere but from `builder()`. Or do you allow people to reference the type and its constructor directly?